### PR TITLE
Clean up pipe pair on transport start failure

### PR DESCRIPTION
**Pull request by mikaelm12**
Completing pipes when our transport fails to start.
Issue: https://github.com/aspnet/SignalR/issues/1733

### CODE REVIEW

File: src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs [hunk 1]
```diff
@@ -356,6 +357,14 @@ private static Uri CreateConnectUrl(Uri url, string connectionId)
             Log.TransportStarted(_logger, _transport);
         }
 
+        private void completePipePair(DuplexPipe.DuplexPipePair duplexPipePair)
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs]: Pass the two pipes in separately. I don't like having `DuplexPipe.DuplexPipePair` here :). Just a style thing.

Comment (mikaelm12) [reply src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs]: Pulled the complete of the pipes out of the method entirely.

File: src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs [hunk 2]
```diff
@@ -345,6 +345,10 @@ private static Uri CreateConnectUrl(Uri url, string connectionId)
             catch (Exception ex)
             {
                 Log.ErrorStartingTransport(_logger, transport, ex);
+                pair.Transport.Input.Complete();
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs]: Add a comment

**Comment by JamesNK** [comment]
Is this something that can be unit tested?

**Comment by davidfowl** [comment]
Yes, making a fake transport that throws in StartAsync then observing that the application pipes are closed. The transport pipe is trickier because it's never exposed and it's nulled out.

**Pull request merged**